platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -27,6 +58,61 @@ - (instancetype)initWithCoordinates:(const CLLocationCoordinate2D *)coords count
     return self;
 }
 
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+    self = [super initWithCoder:decoder];
+    if (self)
+    {
+        NSArray *coordinates = [decoder decodeObjectOfClass:[NSArray class] forKey:@"coordinates"];
+        CLLocationCoordinate2D *coords = (CLLocationCoordinate2D *)malloc([coordinates count] * sizeof(CLLocationCoordinate2D));
+        for (NSUInteger i = 0; i < [coordinates count]; i++)
+        {
+            coords[i] = CLLocationCoordinate2DMake([coordinates[i][0] doubleValue], [coordinates[i][1] doubleValue]);
+        }
+        _coordinates = { coords, coords + [coordinates count] };
+        free(coords);
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder
+{
+    [super encodeWithCoder:coder];
+    
+    NSMutableArray *coordinates = [NSMutableArray array];
+    for (auto coord : _coordinates) {
+        [coordinates addObject:@[@(coord.latitude), @(coord.longitude)]];
```

Comment:
CLLocationCoordinate2D does work with keyed archivers and unarchivers, as long as you use NSValue to wrap it. Archiving a series of CLLocationCoordinate2Ds would be more efficient than archiving a series of dictionaries, and I don't think there's any ambiguity with that approach because we don't define an MGLSwappedCoordinate2D struct anywhere.